react/features/base/conference/middleware.any.ts

```diff
@@ -195,64 +198,68 @@ function _conferenceFailed({ dispatch, getState }: IStore, next: Function, actio
         break;
+
     case JitsiConferenceErrors.REDIRECTED: {
-        // once conference.js is gone this can be removed and both
-        // redirect logics to be merged
-        if (typeof APP === 'undefined') {
-            const newConfig = getVisitorOptions(getState, error.params);
-
-            if (!newConfig) {
-                logger.warn('Not redirected missing params');
-                break;
-            }
+        const newConfig = getVisitorOptions(getState, error.params);
 
-            const [ vnode ] = error.params;
+        if (!newConfig) {
+            logger.warn('Not redirected missing params');
+            break;
+        }
 
-            dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                .then(dispatch(conferenceWillLeave(conference)))
-                .then(conference.leave())
-                .then(dispatch(disconnect()))
-                .then(dispatch(setIAmVisitor(Boolean(vnode))))
+        const [ vnode ] = error.params;
 
-                // we do not clear local tracks on error, so we need to manually clear them
-                .then(dispatch(destroyLocalTracks()))
-                .then(dispatch(connect()));
-        }
+        dispatch(overwriteConfig(newConfig)) // @ts-ignore
+            .then(dispatch(conferenceWillLeave(conference)))
+            .then(conference.leave())
+            .then(dispatch(disconnect()))
+            .then(dispatch(setIAmVisitor(Boolean(vnode))))
+
+            // we do not clear local tracks on error, so we need to manually clear them
+            .then(dispatch(destroyLocalTracks()))
+            .then(() => {
+                dispatch(conferenceWillInit());
+            })
+            .then(dispatch(connect()));
         break;
     }
     }
 
-    if (typeof APP === 'undefined') {
-        !error.recoverable
-        && conference
-        && conference.leave(CONFERENCE_LEAVE_REASONS.UNRECOVERABLE_ERROR).catch((reason: Error) => {
-            // Even though we don't care too much about the failure, it may be
-            // good to know that it happen, so log it (on the info level).
-            logger.info('JitsiConference.leave() rejected with:', reason);
-        });
-    } else {
-        // FIXME: Workaround for the web version. Currently, the creation of the
-        // conference is handled by /conference.js and appropriate failure handlers
-        // are set there.
+    !error.recoverable
+    && conference
+    && conference.leave(CONFERENCE_LEAVE_REASONS.UNRECOVERABLE_ERROR).catch((reason: Error) => {
+        // Even though we don't care too much about the failure, it may be
+        // good to know that it happen, so log it (on the info level).
+        logger.info('JitsiConference.leave() rejected with:', reason);
+    });
+
+    // FIXME: Workaround for the web version. Currently, the creation of the
+    // conference is handled by /conference.js and appropriate failure handlers
+    // are set there.
+    if (typeof APP !== 'undefined') {
```

Comment:
Suggestion:
Maybe we can move this in the middleware.web instead of checking.

Reply:
Yeah we can move it in another PR :)